android/java/org/chromium/chrome/browser/ntp/BraveNewTabPageLayout.java

```diff
@@ -184,9 +184,12 @@ protected int getMaxRowsForMostVisitedTiles() {
     @Override
     protected void onAttachedToWindow() {
         super.onAttachedToWindow();
-        if (sponsoredTab == null)
+        if (sponsoredTab == null) {
             initilizeSponsoredTab();
-        checkAndShowNTPImage(false);
+        }
+        if (sponsoredTab != null) {
```

Comment:
why is this sometimes still null after `initilizeSponsoredTab`?

Reply:
also this logic is repeated in a lot of places, it seems like it should be encapsulated in a method

Reply:
We don't need it, removed in in https://github.com/brave/brave-core/commit/872c78572a41f78df50f4901a6256f7e4608f15d

Reply:
As for logic, it is slightly different. If we want to review it, I believe we should create a follow up issue so @deeppandya could review it later in master.